src/app.js

```diff
@@ -1,13 +1,204 @@
-// Vendor Modules
+// Vendor Modules.
 import $ from 'jquery';
 import _ from 'underscore';
 
-// CSS
+// CSS.
 import './css/foundation.css';
 import './css/style.css';
 
-console.log('it loaded!');
+// Models and collections.
+import Trip from './app/models/trip';
+import TripList from './app/collections/trip_list';
+import Reservation from './app/models/reservation';
+
+// DOM Selectors.
+const $tripsList = $('#trips-list')
+const $tripDescription = $('#trip-description')
+const $reservationTripForm = $('#reservation-trip-form')
+
+// Templates for trip list and details.
+let tripTemplate;
+let tripDetailsTemplate;
+
+// Start of trip list.
+const tripList = new TripList();
+
+const render = function render(tripList) {
+  console.log(tripList);
+  const $trips = $('#trips-list')
+
+  // Remember to empty, so things don't repeat.
+  $trips.empty();
+  tripList.forEach((trip) => {
+    $trips.append(tripTemplate(trip.attributes));
+    console.log(trip.attributes);
+    console.log('It loaded!');
+  });
+};
+
+const fields = ['name', 'id', 'category', 'continent', 'weeks', 'cost', 'about'];
+
+const reservationFields = ['name', 'email', 'age'];
+
+// const updateStatusMessageFrom = (messageHash) => {
+//   $('#status-messages ul').empty();
+//   for(let messageType in messageHash) {
+//     messageHash[messageType].forEach((message) => {
+//       $('#status-messages ul').append($(`<li>${messageType}:  ${message}</li>`));
+//       console.log(`<li>${messageType}:  ${message}</li>`);
+//     })
+//   }
+//   $('#status-messages').show();
+// }
+//
+// const updateStatusMessageWith = (message) => {
+//   $('#status-messages ul').empty();
+//   $('#status-messages ul').append(`<li>${message}</li>`);
+//   $('#status-messages').show();
+// }
+
+const events = {
+  // Add a bloody trip
+  addTrip(event) {
+    event.preventDefault();
+    const tripData = {};
+    fields.forEach( (field) => {
+      const val = $(`input[name=${field}]`).val();
+      if (val != '') {
+        tripData[field] = val;
+      }
+    });
+
+    const trip = new Trip(tripData);
+
+    if (trip.isValid()) {
+      // tripList.add(trip);
+      trip.save({}, {
+        success: events.successfullSave,
+        error: events.failedSave,
+      });
+    } else {
+      console.log('Trip is invalid, sad trombone.');
+      console.log(trip.validationError);
```

Comment:
It's good that you're checking validation errors, but this needs to be reported to the user as well.